crates/oxc_linter/src/rules/eslint/sort_imports.rs

```diff
@@ -0,0 +1,835 @@
+
+                    Fix::new(sorted_text, specifiers_span)
+                },
+            );
+        } else {
+            ctx.diagnostic(sort_members_alphabetically_diagnostic(unsorted_name, unsorted_span));
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct MemberSyntaxSortOrder(Vec<ImportKind>);
+
+impl Default for MemberSyntaxSortOrder {
+    fn default() -> Self {
+        MemberSyntaxSortOrder(vec![
+            ImportKind::None,
+            ImportKind::All,
+            ImportKind::Multiple,
+            ImportKind::Single,
+        ])
+    }
+}
+
+impl MemberSyntaxSortOrder {
+    fn get_group_index_by_import_decl(&self, decl: &ImportDeclaration) -> usize {
+        // import "foo.js" -> ImportKind::None
+        // import * as foo from "foo.js" -> ImportKind::All
+        // import { a, b } from "foo.js" -> ImportKind::Multiple
+        // import a from "foo.js" -> ImportKind::Single
+        // import { a } from 'foo.js' -> ImportKind::Single
+        let import_kind = match &decl.specifiers {
+            Some(specifiers) => {
+                if specifiers.is_empty() {
+                    ImportKind::None
+                } else if specifiers.len() == 1 {
+                    if matches!(
+                        specifiers[0],
+                        ImportDeclarationSpecifier::ImportNamespaceSpecifier(_)
+                    ) {
+                        ImportKind::All
+                    } else {
+                        ImportKind::Single
+                    }
+                } else {
+                    ImportKind::Multiple
+                }
+            }
+            None => ImportKind::None,
+        };
+
+        self.0.iter().position(|kind| kind == &import_kind).unwrap_or_default()
+    }
+    fn get_kind_name_by_index(&self, index: usize) -> Option<&str> {
+        let kind = self.0.get(index)?;
+
+        match kind {
+            ImportKind::None => Some("None"),
+            ImportKind::All => Some("All"),
+            ImportKind::Multiple => Some("Multiple"),
+            ImportKind::Single => Some("Single"),
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone, Hash, Eq, PartialEq)]
+enum ImportKind {
+    // import from 'foo.js'
+    #[default]
+    None,
+    // import * from 'foo.js'
+    All,
+    // import { a, b } from 'foo.js'
+    Multiple,
+    // import a from 'foo.js'
+    Single,
+}
+
+impl FromStr for ImportKind {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "none" => Ok(ImportKind::None),
+            "all" => Ok(ImportKind::All),
+            "multiple" => Ok(ImportKind::Multiple),
+            "single" => Ok(ImportKind::Single),
+            _ => Err("Invalid import kind"),
+        }
+    }
+}
+
+fn get_first_local_member_name(decl: &ImportDeclaration) -> Option<CompactStr> {
+    let specifiers = decl.specifiers.as_ref()?;
+
+    if let Some(first) = specifiers.first() {
+        match first {
```

Comment:
Consider moving this to a new function under `ImportDeclarationSpecifier`'s `impl` so that other rules can benefit from it too.